plugins/statuspage/client.py

```diff
@@ -0,0 +1,107 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Statuspage integration
+
+import requests
+import jsonschema
+import logging
+
+from statuspage import schema
+
+
+class StatuspageClient:
+    def __init__(self, api_key, page_name, group_name):
+        """A Statuspage client for a page and group"""
+
+        self.base_url = "https://api.statuspage.io/v1/"
+        self.api_key = api_key
+
+        self.page_id = self.get_page_id(page_name)
+        self.group_id = self.get_component_group_id(group_name)
+
+    def _request(self, method, path, data=None):
+        headers = {"Authorization": "OAuth {}".format(self.api_key)}
+        request_method = {
+            "get": requests.get,
+            "post": requests.post,
+            "delete": requests.delete,
+            "put": requests.put,
+            "patch": requests.patch,
+        }.get(method)
+
+        if not request_method:
+            raise ValueError("Method {} not supported".format(method))
+
+        url = self.base_url + path
+        resp = request_method(url, json=data, headers=headers)
+        logging.info("status-code {} for {}".format(resp.status_code, url))
+        if resp.status_code != 200:
+            logging.error(resp.content)
+        return resp
+
+    def get_id(self, data, predicate):
+        for row in data:
+            if predicate(row):
+                return row["id"]
+        return None
+
+    def get_page_id(self, name):
+        resp = self._request("get", "pages")
+        data = resp.json()
```

Comment:
Hmm, so for most of these calls, if the response is not 200, we'll probably end up with a `ValueError` exception here, right? That seems like maybe not the ideal place/exception to throw. I'd consider calling `raise_for_status()` above on line 42 instead..

Reply:
I did run into `401`'s and `404`'s while developing the client, Statuspage is good about sending a json response back. It sounds like a good idea to just raise non-200 status up to the operator and airflow for logging.